madgraph/interface/amcatnlo_interface.py

```diff
@@ -119,6 +119,13 @@ def check_display(self, args):
                 args[1] = 'loop' 
 
     def check_add(self, args):
+
+        # this is for the Sudakov approximation of EW corrections
+        logger.warning('Generating a process including the Sudakov approximation of EW corrections.\n' + \
+                       'Please cite arxiv:2110.03714')
+        if "--ewsudakov" in args:
+            self.ewsudakov = True
+            args.remove('--ewsudakov')
         
```

Comment:
can you 
generate a process with sudakov and add without?
Did you check that?

Reply:
yes, acceptance tests all pass.